template/9-design.md

```diff
@@ -2,23 +2,34 @@
 
 ## Frontend
 
-*List the key libraries, languages, components used by the MVP.*
+**Implementation Framework** The application has been developed with Flutter, which is based on the Dart programming language. Flutter can natively compile on Android and iOS from the same code base, which allows us to launch the application to more users easily. 
+We use the Model View ViewModel (MVVM) architecture to ensure structured management and interactions between the UI and the application logic. We use the Riverpod plugin, which simplifies the application of the Observer pattern, reducing necessary UI refreshes.
 
-- Dart
-- Flutter
-- Firebase
-- GeoFlutterFirePlus
+**Firebase communication** We use Firebase Authentication to handle user authentication. The Firebase Core, Firebase Auth and Google Sign-In Flutter plugins control the user authentication flow through a Google Sign-In intent on the Login screen.
 
-*If applicable, describe essential screens.*
+Storing and fetching the user interactions in the application are handled using the Firebase Core and Cloud Firestone Flutter plugins.
 
-- Login
-- Home
-- Profile
-- Post Creation
-- Post Details
-- Challenges
-- Map
+**User location processing** In order to retrieve the user's surrounding posts and display the map with the current user location, we use the GeoLocator Flutter plugin. This plugin simplifies access to the user's current location on every platform. Additionally, we use GeoFlutterFire Plus, which simplifies the storage and fetching of Firebase documents based on their location. 
```

Comment:
You are right: [bf7578e](https://github.com/ProximaEPFL/prd/pull/8/commits/bf7578e083689f6851a53d504e71b08b6fe30e5e)

Note that I made a small mention of geoHashing as we are in the frontend and this may be more suited in the backend 😄 

Tell me what you think

Reply:
Yes great